PR: https://github.com/odoo/odoo/pull/

From: 03aae0d3d5164f00d927e132247a90641ca3b1a0
From: Simon Lejeune
Date: 2016-08-26 13:18:36

Breaking data model changes scores: del:3 + add:0, change matches:
-    popup_content = fields.Html(string="Website Popup Content", translate=True, sanitize=False, default=_default_popup_content)
-    popup_redirect_url = fields.Char(string="Website Popup Redirect URL", default='/')
-    group_website_popup_on_exit = fields.Selection([

Total Changes: 18

[FIX] mass_mailing, website_mass_mailing: "group_website_popup_on_exit" group

This group was defined in the mass_mailing module but is only working when
you install website_mass_mailing. What this group does is allowing to insert
a newsletter snippet in the page, thus it has nothing to do in the mass_mailing
module which doesn't depend on website. Also, the snippet itself was already
defined in website_mass_mailing.

Moved all the things to website_mass_mailing.

================================= pseudo patch: =================================

--- a/addons/mass_mailing/models/mass_mailing.py
+++ b/addons/mass_mailing/models/mass_mailing.py
@@ -30,23 +30,10 @@ class MassMailingList(models.Model):
     _order = 'name'
     _description = 'Mailing List'
 
-    def _default_popup_content(self):
-        return """<div class="modal-header text-center">
-    <h3 class="modal-title mt8">Odoo Presents</h3>
-</div>
-<div class="o_popup_message">
-    <font>7</font>
-    <strong>Business Hacks</strong>
-    <span> to<br/>boost your marketing</span>
-</div>
-<p class="o_message_paragraph">Join our Marketing newsletter and get <strong>this white paper instantly</strong></p>"""
-
     name = fields.Char(string='Mailing List', required=True)
     active = fields.Boolean(default=True)
     create_date = fields.Datetime(string='Creation Date')
     contact_nbr = fields.Integer(compute="_compute_contact_nbr", string='Number of Contacts')
-    popup_content = fields.Html(string="Website Popup Content", translate=True, sanitize=False, default=_default_popup_content)
-    popup_redirect_url = fields.Char(string="Website Popup Redirect URL", default='/')
 
     def _compute_contact_nbr(self):
         contacts_data = self.env['mail.mass_mailing.contact'].read_group([('list_id', 'in', self.ids), ('opt_out', '!=', True)], ['list_id'], ['list_id'])

--- a/addons/mass_mailing/models/res_config.py
+++ b/addons/mass_mailing/models/res_config.py
@@ -13,9 +13,4 @@ class MassMailingConfiguration(models.TransientModel):
         (1, "Allow using marketing campaigns (advanced)")], string="Campaigns",
         implied_group='mass_mailing.group_mass_mailing_campaign',
         help="""Manage mass mailign using Campaigns""")
-    group_website_popup_on_exit = fields.Selection([
-        (0, 'Do not add extra content on website pages to encourage visitors to sign up'),
-        (1, 'Allow the use of a pop-up snippet on website to encourage visitors to sign up on a mass mailing list')
-        ], string="Website Pop-up",
-        implied_group="mass_mailing.group_website_popup_on_exit")
     module_mass_mailing_themes = fields.Boolean("Mass mailing themes")
